mbed_greentea/mbed_test_api.py

```diff
@@ -324,8 +324,9 @@ def get_auto_property_value(property_name, line):
                     output.append('{{mbed_assert}}')
                     break
 
-                # Check for test end
-                if '{end}' in line:
+                # Check for test end. Only a '{{end}}' in the start of line indicates a test end.
+                # A sub string '{{end}}' may also appear in an error message.
+                if re.search('^\{\{end\}\}', line, re.I):
```

Comment:
Thing is that you are checking for ``` ^{{end}}``` not ``` ^{{end}}$``` but I think it makes more sense because everything behind  ```^{{end}}``` might be rubbish.